core/jazz_is-slack-channel-available/test/test.js

```diff
@@ -1,11 +1,282 @@
+      event.method = "";
+      index.genericInputValidation(event)
+        .catch(error => expect(error).to.include({
+          result: 'inputError',
+          message: 'method cannot be empty'
+        }));
+    });
+
+    it("should indicate that method is unsupported if it is not GET", () => {
+      var invalidArray = ["POST", "PUT", "DELETE"];
+      invalidArray.forEach(method => {
+        event.method = method;
+        index.genericInputValidation(event)
+          .catch(error => expect(error).to.include({
+            result: 'inputError',
+            message: 'Unsupported method/request'
+          }));
+      });
+    });
+
+    it("should indicate that query parameter(slack_channel) is missing", () => {
+      event.query = {};
+      index.genericInputValidation(event)
+        .catch(error => expect(error).to.include({
+          result: 'inputError',
+          message: 'Missing input parameter slack_channel'
+        }));
+    });
+
+    it("should indicate unauthorized error", () => {
+      event.principalId = "";
+      index.genericInputValidation(event)
+        .catch(error => expect(error).to.include({
+          result: 'unauthorized',
+          message: 'Unauthorized.'
+        }));
+    });
+
+    it("should successfully validate input data", () => {
+      index.genericInputValidation(event)
+        .then(res => expect(res).to.eq(event.query.slack_channel))
+    });
+  });
+
+  describe('getResponse', () => {
+    var channel_url ,channel_name;
+
+    beforeEach(() => {
+      channel_url = config.public_channel_endpoint;
+      channel_name = event.query.slack_channel;
+    });
+
+    it("should successfully search private slack channel in user's slack list", () => {
+      var resObj = {
+        body: "{\"ok\":true,\"channels\":[{\"id\":\"A1B2C3D4E\",\"name\":\"general\",\"is_channel\":true},{\"id\":\"A1B2C3D4F\",\"name\":\"jazz\",\"is_channel\":true},{\"id\":\"A1B2C3D4G\",\"name\":\"test-channel\",\"is_channel\":true},{\"id\":\"A1B2C3D4H\",\"name\":\"random\",\"is_channel\":true},{\"id\":\"A1B2C3D4I\",\"name\":\"test\",\"is_channel\":true}]}"
+      }
+
+      reqStub = sinon.stub(request, "Request").callsFake(obj => {
+        return obj.callback(null, resObj, resObj.body);
+      });
+      index.getResponse(channel_url, channel_name)
+        .then(res => {
+          expect(res).to.eq('true');
+          sinon.assert.calledOnce(reqStub);
+          reqStub.restore();
+        });
+    });
+
+    it("should successfully search group slack channel in user's slack list", () => {
+      var resObj = {
+        body: "{\"ok\":true,\"groups\":[{\"id\":\"A1B2C3D4E\",\"name\":\"general\",\"is_channel\":true},{\"id\":\"A1B2C3D4F\",\"name\":\"jazz\",\"is_channel\":true},{\"id\":\"A1B2C3D4G\",\"name\":\"test-channel\",\"is_channel\":true},{\"id\":\"A1B2C3D4H\",\"name\":\"random\",\"is_channel\":true},{\"id\":\"A1B2C3D4I\",\"name\":\"test\",\"is_channel\":true}]}"
```

Comment:
Convert this to a readable JSON and then stringify to match the underlying components interface.

Reply:
Updated as per comment